### PR TITLE
Feature/enable zf development mode just like in ZendSkeletonApp

### PR DESCRIPTION
**Pull request by gabbydgab**
Based on my suggestion in https://github.com/zendframework/zend-expressive-skeleton/issues/104

Following changes are:
-  updated composer.json with zfcampus/zf-development-mode and scripts
- added development.config.php.dist file (including .gitignore)
- updated the merged configuration if development-mode is enabled.

### CODE REVIEW

**Comment by gabbydgab** [comment]
Should we include the error_reporting? 

```php
// when all else fails, kindly show me the bugs :)
if ($config['debug']) {
    error_reporting(E_ALL);
    ini_set("display_errors", 1);
}
```

**Purpose:** Making sure that what you commit as production code is clean and Uncle Bob will be happy :+1: :joy:

**Comment by xtreamwayz** [comment]
> Should we include the error_reporting?

You've got whoops to catch errors.

Things that needs to be addressed:

- The config cache is not removed when enabling / disabling debug mode.
- Why is `development.config.php` saved in `./config`? I'd rather see it saved in `./config/autoload/development.local.php`.
- If an user chooses not to install whoops, it shouldn't be forced and left out of development.config.php.dist and composer.json.
- Same goes for minimal setup. Whoops and zf-development-mode should not be installed.

Maybe we should rename the whoops option to `development tools`. Or make whoops an optional package and let the installer handle it (together with the ConfigProvider and component installer).

File: config/config.php [hunk 1]
```diff
@@ -24,6 +24,11 @@
         $config = ArrayUtils::merge($config, include $file);
     }
 
+    // Development mode enabled
+    if (file_exists(__DIR__ . '/../config/development.config.php')) {
+        $config = ArrayUtils::merge($config, require __DIR__ . '/../config/development.config.php');
+    }
```
Comment (weierophinney) [inline config/config.php]: I would place this *after* the block that caches configuration; we likely should *never* cache development configuration.

**Comment by weierophinney** [comment]
I agree with all points that @xtreamwayz notes. If we get those resolved, I'd be happy to incorporate this, however!

**Comment by xtreamwayz** [comment]
@gabbydgab Thanx for this. This feature is included in #54.

**Pull request closed without merging**